tlair/src/test/scala/at/forsyte/apalache/tla/lir/io/TestJsonWriter.scala

```diff
@@ -0,0 +1,475 @@
+  def compareMultiLine(ex: TlaEx, expected: String): Unit =
+    compare(ex, expected, 2)
+
+  test("id") {
+    // awesome
+    compare(
+      name("awesome"),
+      """"awesome""""
+    )
+  }
+
+  test("str") {
+    // "Hello, World!"
+    compare(
+      str("Hello, World!"),
+      """"str:Hello, World!""""
+    )
+  }
+
+  test("int") {
+    // int
+    compare(
+      int(42),
+      """42"""
+    )
+  }
+
+  test("big int") {
+    // int
+    compare(
+      bigInt(BigInt("9876543210")),
+      """"int:9876543210""""
+    )
+  }
+
+
+  test("RealSet") {
+    // Real
+    compare(
+      ValEx(TlaRealSet), // TODO: builders for sets? (Andrey)
+      """"set:Real""""
+    )
+  }
+
+  test("prime name") {
+    // awesome'
+    compare(
+      prime("awesome"),
+      """{"prime":"awesome"}"""
+    )
+  }
+
+  test("empty set") {
+    // { }
+    compare(
+      enumSet(),
+      """{"enum":[]}"""
+    )
+  }
+  //
+  test("singleton set") {
+    // { 42 }
+    compare(
+      enumSet(42),
+      """{"enum":[42]}"""
+    )
+  }
+
+  test("singleton set multi-line") {
+    // { 42 }
+    compareMultiLine(
+      enumSet(42),
+      """{
+        |  "enum": [
+        |    42
+        |  ]
+        |}""".stripMargin
+    )
+  }
+
+  test("enum") {
+    // { 1, 2, 3 }
+    compare(
+      enumSet(int(1), int(2), int(3)),
+      """{"enum":[1,2,3]}"""
+    )
+  }
+
+  test("enum multi-line") {
+    // { 1, 2, 3 }
+    compareMultiLine(
+      enumSet(int(1), int(2), int(3)),
+      """{
+        |  "enum": [
+        |    1,
+        |    2,
+        |    3
+        |  ]
+        |}""".stripMargin
+    )
+  }
+
+  test("tuple") {
+    // << 1, two, "three" >>
+    compare(
+      tuple(int(1), name("two"), str("three")),
+      """{"tuple":[1,"two","str:three"]}"""
+    )
+  }
+
+  test("conjunction") {
+    // a /\ b /\ c
+    compare(
+      and(name("a"), name("b"), name("c")),
+      """{"and":["a","b","c"]}"""
+    )
+  }
+
+  test("minus") {
+    // 1 - 2
+    compare(
+      minus(int(1), int(2)),
+      """{"-":[1,2]}"""
+    )
+  }
+
+  test("function definition") {
+    // [ x \in S, y \in T |-> x + y ]
+    compareMultiLine(
+      funDef(plus("x", "y"), "x", "S", "y", "T"),
+      """{
+        |  "function": {
+        |    "+": [
+        |      "x",
+        |      "y"
+        |    ]
+        |  },
+        |  "where": [
+        |    "x",
+        |    "S",
+        |    "y",
+        |    "T"
+        |  ]
+        |}""".stripMargin
+    )
+  }
+
+  test("function application") {
+    // f[e]
+    compare(
+      appFun("f", "e"),
+      """{"apply":"f","to":"e"}"""
+    )
+  }
+
+  test("operator application") {
+    // A(1,2)
+    compare(
+      OperEx(TlaOper.apply, "A", 1, 2),
+      """{"eval":"A","args":[1,2]}"""
+    )
+  }
+
+  test("double function application") {
+    // f[e][g]
+    compare(
+      appFun(appFun("f", "e"), "g"),
+      """{"apply":{"apply":"f","to":"e"},"to":"g"}"""
```

Comment:
Oh. That is confusing. I did not realize that you reserve `apply` for function application. Shall we call the applications `opApply` and `funApply` respectively?